src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/Regex.cs

```diff
@@ -49,56 +51,11 @@ public class Regex
 
         protected internal RegexRunnerFactory factory;
 
-        internal Dictionary<int, int> _caps;            // if captures are sparse, this is the hashtable capnum->index
-        internal Dictionary<string, int> _capnames;     // if named captures are used, this maps names->index
+        protected internal Hashtable caps;            // if captures are sparse, this is the hashtable capnum->index
+        protected internal Hashtable capnames;     // if named captures are used, this maps names->index
 
-        protected internal String[] capslist;              // if captures are sparse or named captures are used, this is the sorted list of names
+        protected internal string[] capslist;              // if captures are sparse or named captures are used, this is the sorted list of names
         protected internal int capsize;                    // the size of the capture array
-        protected IDictionary Caps
-        {
```

Comment:
The apis (Caps and CapNames) are already shipped in contract 4.1.0.0 and available on Nuget and back ported to Full Framework. These are now redundant given that caps and capnames are being exposed taking dependency on Hashtable. Was api review done to delete these apis? Also these need to be deleted from desktop as well.
cc @weshaggard